pulp_container/app/viewsets.py

```diff
@@ -586,6 +586,65 @@ class ContainerDistributionViewSet(BaseDistributionViewSet):
     queryset = models.ContainerDistribution.objects.all()
     serializer_class = serializers.ContainerDistributionSerializer
     filterset_class = ContainerDistributionFilter
+    permission_classes = (access_policy.DistributionAccessPolicyFromDB,)
+    queryset_filtering_required_permission = "container.view_containerdistribution"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": ["create"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_perms:container.add_containerdistribution",
+                    "has_add_namespace_perms",
+                ],
+            },
+            {
+                "action": ["retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:container.view_containerdistribution",
+                    "has_view_namespace_perms",
+                ],
+            },
+            {
+                "action": ["update", "partial_update"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:container.change_containerdistribution",
+                    "has_view_namespace_perms",
```

Comment:
this is arguable since if i have a view permission on the namespace it does not really mean i can change objects under it.
how about creating a 'has_manipulate_namespace_perms. That would give you permission to update/delete existing distributions under existing namespaces

Reply:
https://github.com/pulp/pulp_container/pull/191#discussion_r544414661